packages/vue-instantsearch/src/components/InstantSearch.js

```diff
@@ -111,6 +116,12 @@ export default createInstantSearchComponent({
     };
   },
   render: renderCompat(function (h) {
+    if (!this.indexName && !this.compositionID) {
+      console.error(
+        'Missing either "index-name" or "compositionID" property on "ais-instant-search"'
+      );
+    }
+
```

Comment:
I couldn't find a way to make either `indexName` or `compositionID` required, so this is the next best thing I could come up with :/